registration/include/pcl/registration/ndt_omp.h

```diff
@@ -0,0 +1,521 @@
+                        double a_t,
+                        double f_t,
+                        double g_t);
+
+  /** \brief Auxiliary function used to determine endpoints of More-Thuente interval.
+    * \note \f$ \psi(\alpha) \f$ in Equation 1.6 (Moore, Thuente 1994)
+    * \param[in] a the step length, \f$ \alpha \f$ in More-Thuente (1994)
+    * \param[in] f_a function value at step length a, \f$ \phi(\alpha) \f$ in More-Thuente (1994)
+    * \param[in] f_0 initial function value, \f$ \phi(0) \f$ in Moore-Thuente (1994)
+    * \param[in] g_0 initial function gradiant, \f$ \phi'(0) \f$ in More-Thuente (1994)
+    * \param[in] mu the step length, constant \f$ \mu \f$ in Equation 1.1 [More, Thuente 1994]
+    * \return sufficient decrease value
+    */
+  inline double
+  auxilaryFunction_PsiMT(double a, double f_a, double f_0, double g_0, double mu = 1.e-4)
+  {
+    return (f_a - f_0 - mu * g_0 * a);
+  }
+
+  /** \brief Auxiliary function derivative used to determine endpoints of More-Thuente interval.
+    * \note \f$ \psi'(\alpha) \f$, derivative of Equation 1.6 (Moore, Thuente 1994)
+    * \param[in] g_a function gradient at step length a, \f$ \phi'(\alpha) \f$ in More-Thuente (1994)
+    * \param[in] g_0 initial function gradiant, \f$ \phi'(0) \f$ in More-Thuente (1994)
+    * \param[in] mu the step length, constant \f$ \mu \f$ in Equation 1.1 [More, Thuente 1994]
+    * \return sufficient decrease derivative
+    */
+  inline double
+  auxilaryFunction_dPsiMT(double g_a, double g_0, double mu = 1.e-4)
+  {
+    return (g_a - mu * g_0);
+  }
+
+  /** \brief The voxel grid generated from target cloud containing point means and covariances. */
+  TargetGrid target_cells_;
+
+  // double fitness_epsilon_;
+
+  /** \brief The side length of voxels. */
+  float resolution_;
+
+  /** \brief The maximum step length. */
+  double step_size_;
+
+  /** \brief The ratio of outliers of points w.r.t. a normal distribution, Equation 6.7 [Magnusson 2009]. */
+  double outlier_ratio_;
+
+  /** \brief The normalization constants used fit the point distribution to a normal distribution, Equation 6.8 [Magnusson 2009]. */
+  double gauss_d1_, gauss_d2_;
+
+  /** \brief The probability score of the transform applied to the input cloud, Equation 6.9 and 6.10 [Magnusson 2009]. */
+  double trans_probability_;
+
+  /** \brief Precomputed Angular Gradient
+   *
+   * The precomputed angular derivatives for the jacobian of a transformation vector, Equation 6.19 [Magnusson 2009].
+   */
+  Eigen::Vector3d j_ang_a_, j_ang_b_, j_ang_c_, j_ang_d_, j_ang_e_, j_ang_f_, j_ang_g_, j_ang_h_;
+
+  /** \brief Matrix composed of j_ang_*_ (for SSE optimization)  */
+  Eigen::Matrix<float, 8, 4> j_ang;
+
+  /** \brief Precomputed Angular Hessian
+   *
+   * The precomputed angular derivatives for the hessian of a transformation vector, Equation 6.19 [Magnusson 2009].
+   */
+  Eigen::Vector3d h_ang_a2_, h_ang_a3_, h_ang_b2_, h_ang_b3_, h_ang_c2_, h_ang_c3_,
```

Comment:
Why not keep them together? aka `A` is a matrix. So are B, C, D,E and F. Would adding 3 zeroes for `a1` make a big impact in performance? The readability would improve by a lot.